### PR TITLE
Clear tornado ioloop before os.fork

### PR DESCRIPTION
**Pull request by yunstanford**
an `IOLoop` cannot survive a fork or be shared across processes in any way. When multiple processes are being used, each process should create its own `IOLoop`. We should probably clear current `IOLoop` if exists before `os.fork`.

### CODE REVIEW

**Comment by tilgovi** [comment]
Thank you for the PR.

The documentation for that method says "Intended primarily for use by test frameworks in between tests." I'm a little concerned that this could mask bugs or create confusion. It also doesn't appear to do anything in the built-in Twisted implementation and [Python has an open bug](https://bugs.python.org/issue21998) for handling `fork()` in asyncio.

I'm not sure what to do.

**Comment by tilgovi** [comment]
I think this is probably safe and maybe good. I'll wait to see if anyone else has thoughts before I merge it, though.

**Comment by yunstanford** [comment]
I can add a comment there, but  yeah, i agree, it probably won't fix all cases,

File: gunicorn/workers/gtornado.py [hunk 1]
```diff
@@ -70,6 +70,12 @@ def heartbeat(self):
                     self.ioloop.stop()
 
     def init_process(self):
+        """\
```
Comment (berkerpeksag) [inline gunicorn/workers/gtornado.py]: Style nit: I'd use `# comment`, not a docstring :)

**Comment by tilgovi** [comment]
Looks good to me. Thank you!

**Pull request merged**